ostree/ostree_transport.go

```diff
@@ -1,242 +1,24 @@
-//go:build containers_image_ostree
-// +build containers_image_ostree
-
 package ostree
 
 import (
-	"bytes"
-	"context"
 	"errors"
-	"fmt"
-	"os"
-	"path/filepath"
-	"strings"
 
-	"github.com/containers/image/v5/directory/explicitfilepath"
-	"github.com/containers/image/v5/docker/reference"
-	"github.com/containers/image/v5/internal/image"
 	"github.com/containers/image/v5/transports"
 	"github.com/containers/image/v5/types"
-	"github.com/containers/storage/pkg/regexp"
 )
 
-const defaultOSTreeRepo = "/ostree/repo"
-
 // Transport is an ImageTransport for ostree paths.
-var Transport = ostreeTransport{}
-
-type ostreeTransport struct{}
-
-func (t ostreeTransport) Name() string {
-	return "ostree"
-}
+//
+// Deprecated: The ostree: implementation has been removed, and any attempt to use this transport fails.
+var Transport = transports.NewStubTransport("ostree")
```

Comment:
What is the point of keeping this? If we don't support any ostree code would not be better to outright cause compile failures for all users? Why would we move all errors to runtime? Sure it break API but removing the actual function already breaks any potential user so I would prefer a proper clean break no?

Reply:
I don’t feel strongly about this.

I think there is, in principle, _some_ value to keeping the transport available, because code matching transports,, similar to https://github.com/containers/common/blob/f71a7a6eca1d1c2846e00b2cda013c9b37af7cc5/libimage/pull.go#L237 , can continue to compile for a while; keeping the symbols around reduces the urgency of updating it.

OTOH it is almost certainly a purely hypothetical situation.

Reply:
Right but I wonder if this not ends up causing more technical debt that way. If nobody updates the places where it is referenced they just end up with defunct code which IMO is worse.

Maybe I do to much rust coding but I like to have changes fail to compile and not find out only at runtime in tests (in the best case) or production (at the worst case) that something changed in an incompatible way.

I don't see any podman references to this the ostree package so I am not that worried about the upgrade path. (I know there are more c/image users so that is an selfish view)
And if things failed to compile already removing it would not make it worse? I guess there is the case about users of the already stub implementation?